starknet_py/net/rpc_schemas/rpc_schemas.py

```diff
@@ -225,26 +228,33 @@ def preprocess(self, data, **kwargs):
 
     @post_load
     def make_dataclass(self, data, **kwargs) -> BlockStateUpdate:
-        storage_diffs, contract_diffs = BlockStateUpdateSchema._extract_diffs(
-            data["state_diff"]
-        )
+        deployed_contracts = data["state_diff"]["deployed_contracts"]
```

Comment:
I wonder if it wouldn't make more sense to declare a separate schema for StateDiff and then have `state_diff = fields.Nested(StateDiffSchema(), ...)`. This way we'd only have to flatten this data in `make_dataclass`, without the need for all of this processing that is error prone.